src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,226 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import java.io.InputStream
+import java.security.MessageDigest
+import java.util.zip.CRC32
+
+/** Asserts that the actual stream has the same content as the other stream.
+ * Both [InputStream]s will be closed by the assertion.
+ *
+ * @param other which content is compared to the actual one
+ */
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+
+    val msg = doTheStreamHaveTheSameContent(actual, other)
+    if (msg != null) {
+        expected(msg)
+    }
+
+}
+
+/** Asserts that the actual stream has a different content as the other stream.
+ * Both [InputStream]s will be closed by the assertion.
+ *
+ * @param other which content is compared to the actual one
+ */
+fun Assert<InputStream>.hasNotSameContentAs(other: InputStream) {
+
+    val msg = doTheStreamHaveTheSameContent(actual, other)
+    if (msg == null) {
+        expected("streams not to be equal, but they were equal")
+    }
+
+}
+
+/** Asserts that compute the MD5 hash of the stream content.
+ *
+ * @param expectedChecksum the expected checksum, every byte is separated by ':'
+ */
+fun Assert<InputStream>.hasMD5(expectedChecksum: String) = hasChecksum("MD5", expectedChecksum)
```

Comment:
How common is checking a hash in this format, or at all for that reason? Wondering if this is generic enough to land in the core lib. You can always have these assertions in your application.

Reply:
In my case I'm testing a REST API and try to stream some data over the write. I want to verify that I got the correct file. Instead of having a copy of the file to use hasSameContentAs I could just use hasMD5, or hasSHA1, or SHA-256 for verification.
Or are you referring to exactly the hasMD5 method?

Reply:
Just thinking this is niche enough that it doesn't deserve to be in the core package.

Reply:
Are you talking about all the checksums or just the md5?

Reply:
All of them.

Reply:
I have a different opinion, but it is your project. I pushed the code without the checksum assertions.

Reply:
Feel free to create a seperate issue for that, I may reconsider if there is enough demand.